src/pharmpy/tools/mfl/parse.py

```diff
@@ -22,4 +34,358 @@ def parse(code: str) -> List[Statement]:
+        if not any(lt in other.lagtime for lt in self.lagtime):
+            name, func = list(self.convert_to_funcs(["lagtime"]).items())[0]
+            lnt[name] = func
+        print(lnt)
+        # TODO : Covariates
+        return lnt
+
+    def stringify(self):
+        return mfl_stringify(self.mfl_statement_list)
+
+    def to_model(self, model: Optional[Model] = None):
+        if not model:
+            model = load_example_model("pheno")
+
+        funcs = self.convert_to_funcs()
+        if len(funcs) > 5:
+            raise ValueError("Object does not represent a single model")
+        for name, func in funcs.items():
+            model = func(model)
+        return model
+
+    def __repr__(self):
+        features = [f'Absorption : {self.absorption}']
+        features.append(f'Elimination : {self.elimination}')
+        features.append(f'Transits : {self.transits} , {self.transits_depot}')
+        features.append(f'Peripherals : {self.peripherals}')
+        features.append(f'Lagtime : {self.lagtime}')
+        features.append(f'Covariate : {self.covariate}')
+        feature_string = "\n".join(features)
+        return f'Modelfeatures\n{feature_string}'
+
+    def __sub__(self, other):
+        return ModelFeatures.create(
+            absorption=tuple([a for a in self.absorption if a not in other.absorption]),
+            elimination=tuple([e for e in self.elimination if e not in other.elimination]),
+            transits=(
+                tuple([t for t in self.transits if t not in other.transits]),
+                tuple([td for td in self.transits_depot if td not in other.transits_depot]),
+            ),
+            peripherals=tuple([p for p in self.peripherals if p not in other.peripherals]),
+            lagtime=tuple([lt for lt in self.lagtime if lt not in other.lagtime]),
+        )
+
+    def __add__(self, other):
+        return ModelFeatures.create(
+            absorption=tuple(
+                list(self.absorption) + [a for a in other.absorption if a not in self.absorption]
+            ),
+            elimination=tuple(
+                list(self.elimination) + [e for e in other.elimination if e not in self.elimination]
+            ),
+            transits=(
+                tuple(list(self.transits) + [t for t in other.transits if t not in self.transits]),
+                tuple(
+                    list(self.transits_depot)
+                    + [t for t in other.transits_depot if t not in self.transits_depot]
+                ),
+            ),
+            peripherals=tuple(
+                list(self.peripherals) + [p for p in other.peripherals if p not in self.peripherals]
+            ),
+            lagtime=tuple(
+                list(self.lagtime) + [lt for lt in other.lagtime if lt not in self.lagtime]
+            ),
+            covariate=self.covariate,  # TODO : Covariates
+        )
+
+    def __eq__(self, other):
+        if (
+            set(self.absorption) == set(other.absorption)
```

Comment:
This might be needed or there might be an invariant hidden that we need to get to the surface.